app/models/submission.rb

```diff
@@ -0,0 +1,21 @@
+class Submission < ApplicationRecord
+  include PublicActivity::Common
+
+  validates_presence_of :description
+  validates_presence_of :state
+
+  belongs_to :fund, -> { where(level: :fund) }, class_name: "Activity"
+  belongs_to :organisation
+
+  validates_uniqueness_of :fund, scope: :organisation
+  validate :activity_must_be_a_fund
+
+  enum state: [:inactive]
+
+  def activity_must_be_a_fund
+    return unless fund
```

Comment:
I found this line confusing - I wonder if `return unless fund.present?` might be more explicit. 

A case of too many `funds` !